aws_lambda_powertools/utilities/batch/base.py

```diff
@@ -146,3 +184,108 @@ def batch_processor(
         processor.process()
 
     return handler(event, context)
+
+
+class BatchProcessor(BasePartialProcessor):
+    DEFAULT_RESPONSE: Dict[str, List[Optional[dict]]] = {"batchItemFailures": []}
+
+    def __init__(self, event_type: EventType, model: Optional["BatchTypeModels"] = None):
+        """Process batch and partially report failed items
+
+        Parameters
+        ----------
+        event_type: EventType
+            Whether this is a SQS, DynamoDB Streams, or Kinesis Data Stream event
+        model: Optional["BatchTypeModels"]
+            Parser's data model using either SqsRecordModel, DynamoDBStreamRecordModel, KinesisDataStreamRecord
+        """
+        self.event_type = event_type
+        self.model = model
+        self.batch_response = self.DEFAULT_RESPONSE
```

Comment:
Hey @heitorlessa , don't know if this class is meant to be extended. Case it's not, following comment won't make much sense but anyways:
Do you think it's worth being more defensive in here by adding a deepcopy ?
```suggestion
        self.batch_response = deepcopy(self.DEFAULT_RESPONSE)
```
Just to avoid a scenario where someone extends from this class and mess with the `DEFAULT_RESPONSE` through `self.batch_response` (`self.batch_response.update({"batchItemFailures": []})` for instance ) - or any sort of changes...